OpenRA.Mods.D2k/Traits/World/BuildableTerrainLayer.cs

```diff
@@ -49,12 +53,27 @@ public void WorldLoaded(World w, WorldRenderer wr)
 		public void AddTile(CPos cell, TerrainTile tile)
 		{
 			map.CustomTerrain[cell] = map.Rules.TileSet.GetTerrainIndex(tile);
+			strength[cell] = info.MaxStrength;
 
 			// Terrain tiles define their origin at the topleft
 			var s = theater.TileSprite(tile);
 			dirty[cell] = new Sprite(s.Sheet, s.Bounds, s.ZRamp, float2.Zero, s.Channel, s.BlendMode);
 		}
 
+		public void HitTile(CPos cell, int damage)
+		{
+			strength[cell] = strength[cell] - damage;
```

Comment:
What happens when `strength[cell] == 0` is true already (the cell doesn't have concrete on it)? I'm not sure how bad counting towards "negative infinity" is in practice, but it might be better to add a safe-guard just in case.